pkg/utils/validation.go

```diff
@@ -0,0 +1,33 @@
+package utils
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+)
+
+// IsOpenshiftSupported will be set to true in case the operator was installed on OpenShift cluster
+var	IsOpenshiftSupported bool
```

Comment:
I'm lazy and let Copilot do the review 😉  The issue is: do not use global variables! 

> I've reviewed the provided code and it appears to be correct. There are no syntax errors, the function ValidateIsOpenshift is properly checking if the cluster has the OpenShift config group, and the global variable IsOpenshiftSupported is being set correctly based on the presence of the OpenShift config group.  However, there is a potential design issue that could be improved. The global variable IsOpenshiftSupported is being set within the ValidateIsOpenshift function. This could lead to unexpected behavior if this function is called multiple times, as the global variable would be overwritten each time. It would be better to return the result from the function and let the caller decide what to do with it. [...]

some more input from my side, skipping answers:
> - please provide an alternative to using the IsOpenshiftSupported global variable, without returning the result in ValidateIsOpenshift
> - make ValidateIsOpenshift private
> - no, isOpenshiftSupported still needs to be public, but not as global variable
> - now call validateIsOpenshift in NewOpenshiftValidator

final code:

```
package utils

import (
 "k8s.io/apimachinery/pkg/runtime/schema"
 "k8s.io/client-go/discovery"
 "k8s.io/client-go/rest"
)

type OpenshiftValidator struct {
 isOpenshiftSupported bool
}

func NewOpenshiftValidator(config *rest.Config) (*OpenshiftValidator, error) {
 v := &OpenshiftValidator{}
 err := v.validateIsOpenshift(config)
 if err != nil {
  return nil, err
 }
 return v, nil
}

func (v *OpenshiftValidator) IsOpenshiftSupported() bool {
 return v.isOpenshiftSupported
}

func (v *OpenshiftValidator) validateIsOpenshift(config *rest.Config) error {
 dc, err := discovery.NewDiscoveryClientForConfig(config)
 if err != nil {
  return err
 }
 apiGroups, err := dc.ServerGroups()
 if err != nil {
  return err
 }

 kind := schema.GroupVersionKind{Group: "config.openshift.io", Version: "v1", Kind: "ClusterVersion"}
 for _, apiGroup := range apiGroups.Groups {
  for _, supportedVersion := range apiGroup.Versions {
   if supportedVersion.GroupVersion == kind.GroupVersion().String() {
    v.isOpenshiftSupported = true
    return nil
   }
  }
 }
 return nil
}
```

Reply:
> do not use global variables

but good idea to run the detection once only 👍🏼